New format of documentation (#369)
husarion/panther_ros: panther_hardware_interfaces/README.md

# panther_hardware_interfaces

Package that implements SystemInterface from ros2_control for Panther.

## ROS Nodes

This package doesn't contain any standalone nodes, only plugins that are loaded by the resource manager. To use this hardware interface, add it to your URDF (you can check how to do it in [panther_description](https://github.com/husarion/panther_ros/tree/ros2/panther_description/)).

> [`PantherSystem`](#panthersystem): Plugin responsible for communicating with engine controllers via the CAN bus, providing E-Stop functionalities and managing Built-in Computer GPIO ports.
> [`PantherImuSensor`](#pantherimusensor): Plugin responsible for communicating with IMU and filtering data using Madgwick Filter.

Reviewer: ```suggestion
- [`PantherSystem`](#panthersystem): Plugin responsible for communicating with engine controllers via the CAN bus, providing E-Stop functionalities and managing Built-in Computer GPIO ports.
- [`PantherImuSensor`](#pantherimusensor): Plugin responsible for communicating with IMU and filtering data using Madgwick Filter.
```
Irrelevant if we decide to move description

# panther_hardware_interfaces

Package that implements SystemInterface from ros2_control for Panther.

## ROS Nodes

> This package contains plugins that are loaded by the resource manager. To use this hardware interface, add it to your URDF (you can check how to do it in [panther_description](https://github.com/husarion/panther_ros/tree/ros2/panther_description/)).

Reviewer: ```suggestion
This package doesn't contain any standalone nodes, only plugins that are loaded by the resource manager. To use this hardware interface, add it to your URDF (you can check how to do it in [panther_description](https://github.com/husarion/panther_ros/tree/ros2/panther_description/)).
```

| Node name          | Description <br/> *Type*                                                                                                                                                    |
| ------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `PantherSystem`    | Plugin responsible for communicating with engine controllers via the CAN bus and providing E-Stop functionalities. <br/> *[`panther_hardware_interfaces/PantherSystem`](.)* |

Reviewer: ```suggestion
| `PantherSystem`    | Plugin responsible for communicating with engine controllers via the CAN bus, providing E-Stop functionalities and managing Built-in Computer GPIO ports. <br/> *[`panther_hardware_interfaces/PantherSystem`](.)* |
```

- `hardware/led_control_enable` [*std_srvs/SetBool*]: Enables or disables SBC (Single Board Computer) control over the LEDs.
- `hardware/motor_power_enable` [*std_srvs/SetBool*]: Enables or disables motor power.

#### Parameters

> Parameters that are required, are defined when including interface in URDF (you can check out [panther_macro.urdf.xacro](../panther_description/urdf/panther_macro.urdf.xacro)).

Reviewer: Thinking about now. Will all of this links work with relative path when parsed to docs in docusaurus?

Author: Yes! ;)
I added this functionality so that we don't have to remember too many things when writing documentation. (the exception are the `<` `>` characters, which are reserved for html.

> [!CAUTION]
> `max_write_pdo_cmds_errors_count`, `max_read_pdo_motor_states_errors_count`, `max_read_pdo_driver_state_errors_count`, `sdo_operation_timeout`, `pdo_motor_states_timeout_ms` and `pdo_driver_state_timeout_ms` are safety-critical parameters, they should be changed only in very specific cases, be sure that you know how they work and be really cautious when changing them.

### PantherImuSensor

#### Parameters

> Parameters that are required, are defined when including interface in URDF (you can check out [panther_macro.urdf.xacro](../panther_description/urdf/panther_macro.urdf.xacro)).

Author: current version is ok
